diktat-gradle-plugin/src/main/kotlin/com/saveourtool/diktat/plugin/gradle/extension/DefaultReporter.kt

```diff
@@ -0,0 +1,163 @@
+/**
+ * All default reporters
+ */
+
+@file:Suppress("UnnecessaryAbstractClass")
+
+package com.saveourtool.diktat.plugin.gradle.extension
+
+import com.saveourtool.diktat.plugin.gradle.defaultReportLocation
+import org.gradle.api.Project
+import org.gradle.api.file.RegularFile
+import org.gradle.api.file.RegularFileProperty
+import org.gradle.api.model.ObjectFactory
+import org.gradle.api.provider.Provider
+import java.io.File
+import javax.inject.Inject
+
+/**
+ * A base interface for reporter
+ *
+ * @param extension extension of generated report
+ * @param objectFactory
+ * @param project
+ * @property id identifier of reporter
+ */
+abstract class DefaultReporter @Inject constructor(
+    val id: String,
+    extension: String,
+    objectFactory: ObjectFactory,
+    project: Project,
+) : Reporter {
+    override val output: RegularFileProperty = objectFactory.fileProperty()
+        .also { fileProperty ->
+            fileProperty.convention(project.defaultReportLocation(extension = extension))
+        }
+}
+
+/**
+ * Plain reporter
+ *
+ * @param objectFactory
+ * @param project
+ */
+abstract class PlainReporter @Inject constructor(
```

Comment:
why should be abstract too?

Reply:
Since gradle creates it and add implementation for managed fields where it's required.
It also can be `open class` and `open fun` -- the main that Gradle should be able to override required methods.

I think abstract class is better (the much better looks when it's an interface, but it doesn't fit for our case)